torch/distributed/_device_mesh.py

```diff
@@ -230,11 +230,10 @@ def _validate_mesh(self):
 
         # validate that all calling ranks pass in the same `mesh` argument.
         self_mesh = self.mesh.to(self.device_type).contiguous()
-        mesh_tensor = funcol.all_gather_tensor(
-            self_mesh, gather_dim=0, group=_get_default_group()
-        )
-        mesh_tensor_chunked = torch.chunk(mesh_tensor, get_world_size())
-        for other_rank, other_mesh in enumerate(mesh_tensor_chunked):
+        tensor_list = [torch.zeros_like(self_mesh) for _ in range(get_world_size())]
+        all_gather(tensor_list, self_mesh)
+
+        for other_rank, other_mesh in enumerate(tensor_list):
```

Comment:
can we also turn off `_validate_mesh` in `init_device_mesh`?

Reply:
> can we also turn off `_validate_mesh` in `init_device_mesh`?

Ye. Will do it in a separate PR. I am trying this out in a unit test and the behavior is not as I expected. So gonna do a follow up PR for this.